libraries/chain/include/eosio/chain/hotstuff.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+#include <eosio/chain/block_header.hpp>
+#include <fc/bitutil.hpp>
+#include <fc/crypto/bls_private_key.hpp>
+#include <fc/crypto/bls_public_key.hpp>
+#include <fc/crypto/bls_signature.hpp>
+#include <fc/crypto/bls_utils.hpp>
+
+namespace eosio::chain {
+
+   const block_id_type NULL_BLOCK_ID = block_id_type("00");
+   const fc::sha256 NULL_PROPOSAL_ID = fc::sha256("00");
+
+   inline uint64_t compute_height(uint32_t block_height, uint32_t phase_counter) {
+      return (uint64_t{block_height} << 32) | phase_counter;
+   }
+
+   struct extended_schedule {
+      producer_authority_schedule                          producer_schedule;
+      std::map<name, fc::crypto::blslib::bls_public_key>   bls_pub_keys;
```

Comment:
Should the `bls_pub_keys` be added in the `producer_authority_schedule` instead of on the side like here?

Reply:
Extended schedule was a placeholder until we reached a decision on the final format of the finalizers set structure. I think we agreed to only keep the bls public key, as well as the threshold for finalizers (as far as the hotstuff algorithm is concerned). The name would be present in the structure as well, but serves only to identify the finalizer on-chain, and is not required by Hotstuff.